disk-utils/mkfs.bfs.c

```diff
@@ -260,13 +264,29 @@ int main(int argc, char **argv)
 	sb.s_start = cpu_to_le32(ino_bytes + sizeof(struct bfssb));
 	sb.s_end = cpu_to_le32(total_blocks * BFS_BLOCKSIZE - 1);
 	sb.s_from = sb.s_to = sb.s_backup_from = sb.s_backup_to = -1;
-	memcpy(sb.s_fsname, fsname, 6);
-	memcpy(sb.s_volume, volume, 6);
+
+	if (fsname == NULL){
+		fsname = "      ";
+		memcpy(sb.s_fsname, fsname, 6);
+	}
+	else{
+		memcpy(sb.s_fsname, fsname, 6);
+		free(fsname);
```

Comment:
This is incorrect, and the original version was also incorrect. You cannot blindly copy 6 bytes; the getopt_long() block checks for `if (len <=0 || len > 6)`.  What if `fsname` is only 4 bytes (e.g. "foo\0")?